### ISSUE TITLE
Disable automatic column selection

### ISSUE THREAD

**Issue by jshearer** [mikro-orm/mikro-orm#427]
**Is your feature request related to a problem? Please describe.**
I'm trying to disable the selection of a specific column from queries by default. My specific usecase is a large text column that I'd love to only query when I really need to, though users in the TypeORM repo have [talked](https://github.com/typeorm/typeorm/issues/828) [about](https://github.com/typeorm/typeorm/issues/440) [wanting](https://github.com/typeorm/typeorm/issues/2819) to prevent selecting a `password` column (though they really shouldn't be storing passwords in plaintext anyway so I'm not sure why this matters).

**Describe the solution you'd like**
Initially I looked to see if I could pass a `selected: false` property to `@Property`, but then I realized that this would mean that I wouldn't actually have a good way to access the column when I want it. I believe TypeORM's solution is to transparently turn the field into an async-getter (i.e a getter that returns a Promise), which... might be a solution here? Something like this?

```es6
@Property({select: false})
long_text: Promise<string>;
```
Though working with raw promises like that seems somewhat confusing (does the promise run when the object is loaded, or just when I await it?), and doesn't seem coherent with the design of this library.

One thought I had is to treat this similarly to `Reference` fields (or maybe a special case `Reference`?), and in fact, I considered just making this a one-to-one reference: when I want to load `long_text`, I can just call `await long_text.load()`.

**Comment by B4nan** [comment]
Having simple 1:1 relation is something I would do myself (with current orm version), but that is a bit different use case where you are designing the schema yourself, not when you work with an existing one. 

You could also use `fields` option (`FindOptions/FindOneOptions`) to specify what you want to load, but it's a positive list, not list of exceptions. Although you could use entity metadata to construct the list of properties quite easily:

```typescript
const meta = em.getMetadata().get('Author');
const blacklist = ['longText', ...]; // names of actual properties in entity definition, not table columns
const fields = Object.keys(meta.properties).filter(prop => !blacklist.includes(prop));
const authors = await em.find(Author, { ... }, { fields });
```

Sure we can add `select: false` to ease this. I am not sure about the promise getter idea though. It could work but it would produce 1 query for 1 entity. Will need to experiment a bit with that.

**Comment by jshearer** [comment]
Since I am designing this myself I ended up going with the 1:1 approach and it's working fine.

I also agree about the promise getter. Having `select: false` influence the default EntityManager `fields` would be neat. I guess this would conceptually be similar to how `eager` works, but for columns instead of relations (and having the opposite effect).

**Comment by B4nan** [comment]
Closing as implemented in `dev` branch via #585, subscribe to #527 for updates.
